engines/sci/graphics/macfont.h

```diff
@@ -0,0 +1,104 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SCI_GRAPHICS_MACFONT_H
+#define SCI_GRAPHICS_MACFONT_H
+
+#include "common/hashmap.h"
+#include "sci/graphics/helpers.h"   // for GuiResourceId
+
+namespace Common {
+class MacResManager;
+}
+namespace Graphics {
+class MacFontManager;
+class Font;
+}
+
+namespace Sci {
+
+/**
+ * GfxMacFontManager handles loading Mac fonts and mapping them to SCI fonts.
+ *
+ * Most Mac SCI1/1.1 games use native Mac fonts to draw their controls.
+ * This was done by altering the kernel functions used by the control scripts
+ * such as kDrawControl and kTextSize. All dialog text and buttons use these
+ * functions so this affects most text in SCI games. Scripts that draw text with
+ * kDisplay are unaffected and continue to use SCI fonts.
+ *
+ * The Mac game window could be set to three sizes: 100% (small), 150% (medium),
+ * or 200% (large). These percentages were relative to the internal resolution
+ * of the game; usually 320x200. The game's screen was stretched to the window
```

Comment:
Would it be possible for this to be configurable for the benefit of platforms that don't support 640x480 games?

Reply:
Is there a way to query if a platform supports 640?

Reply:
I'm investigating wiring this up to the existing "enable_high_resolution_graphics" config value. If it can be done with the default behavior I want then it sounds good. Or, if the small Mac font doesn't work well, we can just go back to SCI font in low-res mode like now.

Reply:
It's done, I've added the existing "Enable high resolution graphics" config setting to these games. If disabled then the small Mac fonts are used with no upscaling. Great suggestion, and it turned up a bug in the low-res iconbar code. Thanks for reviewing all of this!

Reply:
Great! :)